eth/embobj/plus/comm-v2/icub/EoUpdaterProtocol.h

```diff
@@ -63,16 +63,18 @@ typedef enum
     eLoader                 = 0,    /**< the eLoader: the one executed just after reset which loads any other eProcess */
     eUpdater                = 1,    /**< the eUpdater: the one responsible for performing FW update */
     eApplication            = 2,    /**< the eApplication: the standard application */
-    eApplPROGupdater        = 3,    /**< the application used to perform programming of a new eUpdater */
+    eOther01                = 3,    /**< the eOther01: initially the application running on cm4 of AMC board */
+    eApplPROGupdater        = 4,    /**< the application used to perform programming of a new eUpdater */
```

Comment:
better to keep ` eApplPROGupdaterc=3`.

Reply:
Yes I could revert it, but in `icub-firmware\emBODY\eBcode\arch-arm\libs\highlevel\services\embenv\api\eEcommon.h` there is an enum:

![image](https://user-images.githubusercontent.com/18039045/212779625-e8fb71cb-3864-4d31-a675-c7db0e58cced.png)

Which is used by the `updater-core.c` and `updater-embot-main.cpp`

A possible solution would be to add a new `ee_proc_PROGupdater` in the `eEprocessvalues_t` enum with value 3 (the same we find in `icub-main` originally), and continue with `ee_procOther01` with value 4 and so on.

What do you think?

Reply:
The value `eApplPROGupdater` is better not to be changed. It may break compatibility between `FirmwareUpdater` and some legacy binaries. 

If we change its value we should recompile, test and publish many binaries for the `ems`, `mc4plus`, `mc2plus`.
 
For instance, when we want to update the `eUpdater`, we load a special application called [emsAplPROGupdater.hex](https://github.com/robotology/icub-firmware-build/tree/master/ETH/EMS/bin/environment). This binary replies to the message `eOuprot_cmd_DISCOVER_REPLY_t` sent by the `FirmwareUpdater` with the following code:

![image](https://user-images.githubusercontent.com/7148284/213685927-780ef3bb-7049-4c27-8df5-32600b892d45.png)

where:

![image](https://user-images.githubusercontent.com/7148284/213686058-53d06cbf-0fe8-4522-baca-20ed49b9456b.png)

So, with the value = 3.

If we change `eApplPROGupdater` to be = 4 and and we don't recompile, test, release also in the binary, ... the newly compiled `FirmwareUpdater` will not correctly detect what is running on the legacy ETH boards.

That is why I tell is better not to change it.

Reply:
Moreover, about the following:

> A possible solution would be to add a new `ee_proc_PROGupdater` in the `eEprocessvalues_t`  enum with value 3 (the same we find in `icub-main` originally), and continue with `ee_procOther01` with value 4 and so on.

The `eEprocessvalues_t` contains teh places where teh binary stay inside the FLASH and both `eApplication` and `eApplPROGupdater` stays in the same FLASH partition identified by `ee_procApplication`.

 So, I think this enum should not be changed.

Reply:
At this point, how to solve?

Can't we just use the following?

```C++
eOther01                = 4,    /**< the eOther01: initially the application running on cm4 of AMC board */
```

Try that and then we can align ourselves f2f in front of the code.

Reply:
I did it as suggested by you in the last comment, and it works!